### PR TITLE
Adding DevTest Lab Artifact facts

### PR DESCRIPTION
**Pull request by zikalino**
##### SUMMARY
Adding artifact facts module.

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
azure_rm_devtestlabartifact_facts

### CODE REVIEW

**Comment by Fred-sun** [comment]
@yungezz @yuwzho

**Comment by yungezz** [comment]
is this duplicate with artifactsource_facts?

**Comment by zikalino** [comment]
@yungezz no, it's not.
azure_rm_artifactsource_facts lists artifact sources.
while this one lists artifacts within artifact source.
it's actually pretty useful as it allows users to check what DLT has found in the GitHub repo, and how it translates to IDs etc.

File: lib/ansible/modules/cloud/azure/azure_rm_devtestlabartifact_facts.py [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+
+class AzureRMArtifactFacts(AzureRMModuleBase):
+    def __init__(self):
+        # define user inputs into argument
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            lab_name=dict(
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_devtestlabartifact_facts.py:152]: lab_name not required?

Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_devtestlabartifact_facts.py:152]: indeed, I wonder why it wasn't caught by sanity tests.....

**Pull request merged**